c2rust-analyze/src/context.rs

```diff
@@ -945,9 +945,12 @@ impl<'a, 'tcx> AnalysisCtxt<'a, 'tcx> {
         }
     }
 
-    pub fn type_of_rvalue(&self, rv: &Rvalue<'tcx>, loc: Location) -> LTy<'tcx> {
+    /// Returns the [`LTy`] of an [`Rvalue`] and a boolean indicating whether or
+    /// not the `Rvalue` contains a field projection.
```

Comment:
Can you clarify the precise meaning of "the `Rvalue` contains a field projection"?  Which of these count as containing a field projection?

```Rust
let r1 = x.field;
let r2 = &x.field;
let r3 = &(*p).field;
let r4 = x.field + y;
// ^ Rvalue::Add(Operand::Copy(x.field), ...)
```